### ISSUE TITLE
Markdown file names should more closely follow document section names

### ISSUE THREAD

**Issue by ahouseholder** [CERTCC/SSVC#254]
The names of files in the [`/doc/md_src_files`](../blob/main/doc/md_src_files) are largely a historical accident and are increasingly unmoored from the chapter/section headings they contain. The net result is that it can be difficult to translate starting from a specific section in the `.pdf` or `.html` rendering to which `.md` file one can expect to find the content in. The finished product is of course a holistic `.pdf` or `.html` document, whereas the `.md` files are primarily segmented for the convenience of authors and editors.

For example, as of the date of this ticket, there is no obvious way to infer from looking at a heading in the `.pdf` like *7.4.1.3 Overfitting* that this section is contained in the file named [`060_decision-trees.md`](../blob/main/doc/md_src_files/060_decision-trees.md). 

This forces anyone making changes to search the repository first, which adds friction to the editing process.

The build process as in [/.github/workflows/pandoc_html_pdf.yaml](../blob/main/.github/workflows/pandoc_html_pdf.yaml) imposes a constraint on the `.md` file names such that they **must** be ASCII-sortable. Hence we have adopted a three-digit numeric prefix to ensure the files are sorted correctly when passed into `pandoc` to generate the `.pdf` and `.html` documents.

So we need a solution that:

1. Preserves the ASCII sortability feature
2. Better reflects the document structure in the `.md` file name

I believe the following addresses both requirements.
My proposal is to rename all the files in [`/doc/md_src_files`](../blob/main/doc/md_src_files) using the following pattern: `CC_SS_NN_header.md` where:
- `CC` = two-digit numeric chapter number
- `SS` = two-digit numeric section number (or literally `00` if this file begins at a chapter)
- `NN` = two-digit subsection number (or literally `00` if the file begins at a chapter or section)
- `header` = a string that closely resembles or matches the first heading in the file

Assumptions:

1. Each file begins at a heading of either `#` (Chapter), `##` (Section), or `###` (Subsection).
2. No file begins at a sub-subsection (`####`)
3. The document has fewer than 99 chapters
4. No chapter has more than 99 sections
5. No section has more than 99 subsections

This will also allow us to possibly adjust the file breakpoints more easily in the future and retain some logical continuity between the structure of the finished document and its component source `.md` files.

**Comment by ahouseholder** [comment]
Note that because of the number of file renames involved, this change, if implemented, should be done when the pull request queue is otherwise empty w.r.t. changes to the `md_src_files` directory because any PR that implements this suggested change _will_ cause merge conflicts with any other in-flight `.md` changes.

**Comment by j---** [comment]
Another assumption is that it's OK to rename files if a chapter is introduced before another chapter, a section in a chapter is introduced before an existing section in that chapter, etc. 
I think that's an acceptable assumption. The wonky three digit numbering scheme that i was arbitrarily using was partially an attempt to avoid file renaming by just spacing the numbers out to allow space for new files to get interleaved without renaming everything within the chapter. 

If we do this, I also suggest making perhaps more files? So each file is smaller and sections are easier to find? Such as perhaps each section gets its own file?

Are there any sections that are so big they need multiple files? The biggest chapter is *Likely Decision Points and Relevant Data*, but I don't think any of its sections would need to be in multiple files. This suggests `CC_SS` might be an adequate prefix. Though no harm is giving space to grow with `CC_SS_NN`

**Comment by ahouseholder** [comment]
Turns out we don't currently need `_NN` so in #263 I just went with the `CC_SS_` prefix _where necessary_. Chapters that are short enough not to need split up just use a `CC_` prefix.

In the future, should it become necessary to split a section into subsection files, we can follow the same pattern:
- `CC_SS_foo.md` will become `CC_SS_00_foo.md`, `CC_SS_01_foo_first_section_name.md` etc. 

I.e., we can invoke YAGNI and only add another prefix level specifically where and when we need it.
